pkg/resourcekeeper/delete.go

```diff
@@ -85,7 +86,10 @@ func (h *resourceKeeper) delete(ctx context.Context, manifest *unstructured.Unst
 		}
 	}
 	// 2. delete manifests
-	if err = h.Client.Delete(multicluster.ContextWithClusterName(ctx, oam.GetCluster(manifest)), manifest); err != nil && !kerrors.IsNotFound(err) {
+	deleteCtx := multicluster.ContextWithClusterName(ctx, oam.GetCluster(manifest))
+	deleteCtx = oamutil.SetServiceAccountInContext(
```

Comment:
```suggestion
	deleteCtx = oamutil.SetServiceAccountInContext(deleteCtx, h.app)
```

Reply:
The logic of getting the ServiceAccount name and namespace converges in the method.

Reply:
I think passing app directly is less extensible than passing the namespace / serviceAccountName pair. Therefore, I added `GetServiceAccountName`, which extracts ServiceAccount name from annotation like below:

https://github.com/oam-dev/kubevela/pull/3434/commits/78bc21b943410ccaeb4d955d146ea3adb730b4f7#diff-6740e9863579149c3533908aaffc41f207ae7a1031759e1ac075c703f74b74ec

and make all callers use it. WDYT?

Reply:
But, the namespace is also not fixed to `app.namespace`

Reply:
> But, the namespace is also not fixed to app.namespace

Could you give me more details or additional context for this comment?

Reply:
We need to get the ServiceAccount name and namespace from the application, Currently from annotations, and possibly from the spec in the future. 

In multi-cluster mode, the name and namespace of ServiceAccount are both possible to define in a spec.

Reply:
Thanks for the kind explanation 🙂 

If so, both `GetServiceAccountName` and passing application might be ambiguous since multi-cluster mode may contain multiple ServiceAccount names per cluster.

For making the context clear, how about changing `GetServiceAccountName` to `GetServiceAccountNameFromAnnotation` and letting method callers select which ServiceAccount or namespace to use?

Reply:
Agree to this change.